pogi7/luminosityleds
# LL-121: Login user

# Description

This MR adds login functionality to allow users to login into the website.  If users aren't registered then the users won't be able to login into the website. 

https://luminosity-led.atlassian.net/browse/LL-121

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

Test google login should be passing - https://github.com/pogi7/luminosityleds/pull/47/checks

Front-end launch instructions:
1. `cd client`
2. Get mongodb creds from me, Aaron, or Jason
3. Export these creds as environmental variables in your shell
4. `npm run start`

Back-end launch instructions:
1. Open up a new terminal
2. Get mongodb creds from me, Aaron, or Jason
3. Export these creds as environmental variables in your shell
4. `cd server`
5. `npm run serverStart`

Login Instructions:
1. Go to http://localhost:3000/#/login
2. Login with your google account, if you don't have an account you'll get be redirected to register an account.
3. If you already have an account, you should be logged in



- [ ] Launch the front-end and back-end
- [ ] Click the Log In button in the upper right corner
- [ ] Click the `Login with Google` button
- [ ] If you're registered with Luminosity LEDs, you should now be able to login
- [ ] If you're not registered, click the `Register Here` button to register an account

# Checklist (when relevant):


    [TypeScript](https://google.github.io/styleguide/tsguide.html)
[Python](https://peps.python.org/pep-0008/)
[Schema](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/forefront-2010/ee652242(v=vs.100))
- [ ] I have included a reviewer on the pull request [*Replace this text with the reviewer*]
